lib/src/wallet/transactions.rs

```diff
@@ -213,33 +219,52 @@ impl WalletTxns {
         &self.last_txid
     }
 
-    pub fn get_notes_for_updating(&self, before_block: u64) -> Vec<(TxId, SaplingNullifier)> {
+    pub fn get_notes_for_updating(&self, before_block: u64) -> Vec<(TxId, WalletNullifier)> {
         let before_block = BlockHeight::from_u32(before_block as u32);
 
         self.current
             .iter()
             .filter(|(_, wtx)| !wtx.unconfirmed) // Update only confirmed notes
             .flat_map(|(txid, wtx)| {
                 // Fetch notes that are before the before_block.
-                wtx.sapling_notes.iter().filter_map(move |snd| {
-                    if wtx.block <= before_block
-                        && snd.have_spending_key
-                        && snd.witnesses.len() > 0
-                        && snd.spent.is_none()
-                    {
-                        Some((txid.clone(), snd.nullifier.clone()))
-                    } else {
-                        None
-                    }
-                })
+                wtx.sapling_notes
+                    .iter()
+                    .filter_map(move |snd| {
+                        if wtx.block <= before_block
+                            && snd.have_spending_key
+                            && snd.witnesses.len() > 0
```

Comment:
Why is this necessary?

Reply:
Can we make `snd` and `ond` more explicit?
Are they `$DOMAIN_note_descriptors`?

Reply:
> Can we make `snd` and `ond` more explicit? Are they `$DOMAIN_note_descriptors`?

They are, I can change the names, yes.